index.d.ts

```diff
@@ -0,0 +1 @@
+export default function<T extends { [key: string]: any }>(str: string, mix: T): string;
```

Comment:
```suggestion
export type Values = Record<string, any> | any[];
export default templite<T extends Values>(template: string, values: T): string;
```

[Playground](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBqBDANgV2AZzgF44AlYAY2gBMAefGKASwDsBzAGjgxcQD44AH268A2gF0A3AChqlLBiioAZjhYUYTCCwTAAtmCxMYwWgBU4oEy2qFMuAnwAUJgwpMAuOA2bsuAN2w8fC8zAEovH1Y2GWlXQ2NgJwByAAlgLCwILgBvHJYMPWAAX2KAQmTcuAKir2SAd2gsamS4YrCZAHpOogF0zOy4Rqhmsuk4-QSTFNSIeupEXJyABlKyuDyARlLKuFFksEUYFmAoXeTAPg3AAl3k8Q7pbt64WfnFuEPYE6h16-H4o2myWUEAgXgARHlgRBSmDJHAAEaKcF5RFQGGSXY5OBQrybABMAGY2vdHgIcXAwfiCbCEUiKbC-pMAUkAAbSOBwAByhWAXjyNWAADoFAxSksBYLlEwoKLiuy4AAZCAUDCabR8nIYajUJT4fCCijGRClOBOPJanUEfVUNSMY3teWzeHwpgEDUACwgztd+tWxSWnu9BEF239GxygZdwbxpWkLNy8oFfPlHKlMpgdQVOAA1sBKim4CKM3BkgBRaj1RS2ZLy-3yi26kIbAuGpCZiCEACC7AyBHzHI5NpYdrqAFUAMqdmscuscyM+rz7YCq874LDAYBgc5KFAr8TSdpdHoCbm1ODlytQWxcLO5h7HxXK1VaFheJVdnvrwhOCedsL3p4nSjJtlxgLg1w3MAuB3UCgA)